atlasdb-config/src/test/java/com/palantir/atlasdb/factory/TransactionManagersTest.java

```diff
@@ -440,6 +441,65 @@ public void keyValueServiceMetricsDoNotContainUserAgent() {
                 .anyMatch(metricName -> metricName.contains(USER_AGENT_NAME)), is(false));
     }
 
+    @Test
+    public void serviceNameIsFetchedFromAtlasConfig() {
+        KeyValueServiceConfig kvs = new InMemoryAtlasDbConfig();
+        AtlasDbConfig atlasDbConfig = ImmutableAtlasDbConfig.builder()
+                .keyValueService(kvs)
+                .namespace(Optional.of("namespace"))
+                .build();
+        MetricRegistry metrics = new MetricRegistry();
+        TransactionManagers transactionManagers = TransactionManagers.builder()
+                .config(atlasDbConfig)
+                .userAgent(USER_AGENT)
+                .globalMetricsRegistry(metrics)
+                .globalTaggedMetricRegistry(DefaultTaggedMetricRegistry.getDefault())
+                .registrar(environment)
+                .build();
+
+        assertThat(transactionManagers.getServiceName()).isEqualTo("namespace");
+    }
+
+    @Test
+    public void serviceNameIsFetchedFromKvsConfigWhenItIsNotPresentInAtlasConfig() {
+        KeyValueServiceConfig kvs = mock(KeyValueServiceConfig.class);
+        when(kvs.type()).thenReturn("memory");
+        when(kvs.namespace()).thenReturn(Optional.of("namespace"));
+        when(kvs.concurrentGetRangesThreadPoolSize()).thenReturn(1);
+
+        AtlasDbConfig atlasDbConfig = ImmutableAtlasDbConfig.builder()
+                .keyValueService(kvs)
+                .build();
+        MetricRegistry metrics = new MetricRegistry();
+        TransactionManagers transactionManagers = TransactionManagers.builder()
+                .config(atlasDbConfig)
+                .userAgent(USER_AGENT)
+                .globalMetricsRegistry(metrics)
+                .globalTaggedMetricRegistry(DefaultTaggedMetricRegistry.getDefault())
+                .registrar(environment)
+                .build();
+
+        assertThat(transactionManagers.getServiceName()).isEqualTo("namespace");
+    }
```

Comment:
We might want a timelock one for completeness?

Reply:
This is hard to write because the migrator tries to reach to timeLock servers.